src/main/java/com/leverx/employeestat/rest/service/impl/ReportServiceImpl.java

```diff
@@ -32,45 +53,111 @@ public ReportServiceImpl(WorkRepository workRepository) {
     }
 
     @Override
-    public void export(HttpServletResponse response) {
-        createHeaders();
-        try (ServletOutputStream outputStream = response.getOutputStream()) {
+    @Scheduled(cron = "0 0 0 1 * *")
+    @Transactional
+    public void generateAvailableEmployeesReport() {
+        createTableHeaders(FIRST_NAME_COL, LAST_NAME_COL, DEPARTMENT_COL);
+        List<Employee> availableEmployees = workRepository.findAllAvailableNext(30)
+                .stream()
+                .map(Work::getId)
+                .map(WorkId::getEmployee)
+                .collect(Collectors.toList());
+
+        for (int i = 0; i < availableEmployees.size(); i++) {
+            Employee employee = availableEmployees.get(i);
+            Row row = sheet.createRow(i + 1);
+
+            Cell firstName = row.createCell(0);
+            firstName.setCellValue(employee.getFirstName());
+
+            Cell lastName = row.createCell(1);
+            lastName.setCellValue(employee.getLastName());
+
+            Cell department = row.createCell(2);
+            department.setCellValue(employee.getDepartment().getName());
+        }
+
+        File file = new File(String.format("%s/AvailableEmployees-%s.xlsx", reportsPath, LocalDate.now()));
+        try (FileOutputStream outputStream = new FileOutputStream(file)) {
             workbook.write(outputStream);
         } catch (IOException e) {
-            System.out.println("AAAA");
+            throw new ReportWritingException("Failed to write report", e);
         }
     }
 
-
-    private void fillTable() {
+    @Override
+    @Transactional
+    public void exportOccupationReport(HttpServletResponse response) {
```

Comment:
I don't think that service should be aware about HttpServletResponse and directly write to it. Its a mixture of layers and also service can't be tested standalone without http context. I would suggest extracting writing report to the response to the controller at least. In this case it is not a business logic, it is just http response creation so it is okay to put it to the controller.